ignite/contrib/handlers/tensorboard_logger.py

```diff
@@ -419,20 +426,105 @@ class WeightsHistHandler(BaseWeightsHistHandler):
+    def __init__(
+        self,
+        model: nn.Module,
+        tag: Optional[str] = None,
+        whitelist: Optional[Union[List[Union[str, nn.Module]], Callable[[str, nn.Parameter], bool]]] = None,
+    ):
         super(WeightsHistHandler, self).__init__(model, tag=tag)
 
+        self.weights = None
+        if whitelist is None:
+            return
+
+        self.weights = {}
+        if callable(whitelist):
+            for n, p in model.named_parameters():
+                if whitelist(n, p):
+                    self.weights[n] = p
+
+            if len(self.weights) == 0:
+                warnings.warn("Given callable whitelist does not " "select any parameter to be logged.")
+            return
+
+        for item in whitelist:
+            if isinstance(item, str):
+                try:
+                    for n, p in model.get_submodule(item).named_parameters(prefix=item):
+                        self.weights[n] = p
+                except AttributeError:
+                    try:
+                        self.weights[item] = model.get_parameter(item)
+                    except AttributeError:
+                        try:
+                            _ = model.get_buffer(item)
+                            raise ValueError(
+                                "Whitelist weights should not be Buffers as they are"
+                                f" considered persistent. Given buffer name `{item}`"
+                            )
+                        except AttributeError as e:
+                            raise ValueError(
+                                f"Whitelist weight `{item}` is not among model's parameters or submodules"
+                            ) from e
+            elif isinstance(item, nn.Module):
+                name = None
+                for n, m in model.named_modules():
+                    if m == item:
+                        name = n
+                        break
+                if name is not None:
+                    for n, p in item.named_parameters(prefix=name):
+                        self.weights[n] = p
+                else:
+                    raise ValueError(f"Module {item} is not among model's submodules")
+            else:
+                raise ValueError(f"Whitelist weights shoud be string or nn.Module, given {type(item)}")
+
     def __call__(self, engine: Engine, logger: TensorboardLogger, event_name: Union[str, Events]) -> None:
         if not isinstance(logger, TensorboardLogger):
             raise RuntimeError("Handler 'WeightsHistHandler' works only with TensorboardLogger")
 
         global_step = engine.state.get_event_attrib_value(event_name)
         tag_prefix = f"{self.tag}/" if self.tag else ""
-        for name, p in self.model.named_parameters():
-            if p.grad is None:
-                continue
+        weights = self.weights.items() if self.weights is not None else self.model.named_parameters()
```

Comment:
Can't we define `self.weights = self.model.named_parameters()` in the constructeur if `whitelist is None` instead of here ?